From e4cba32b28bac042e53fc6b8a1a9e4126f50f221 Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jekoritz@microsoft.com>
Date: Thu, 7 Mar 2024 14:39:30 -0800
Subject: [PATCH] Forward the cross argument from the outer build to the inner
 build

Backport: TBD
---
 eng/DotNetBuild.props | 1 +
 1 file changed, 1 insertion(+)

diff --git a/eng/DotNetBuild.props b/eng/DotNetBuild.props
index a6350c7fea9..127d6a5d12b 100644
--- a/eng/DotNetBuild.props
+++ b/eng/DotNetBuild.props
@@ -54,6 +54,7 @@
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)nodereuse $(ArcadeFalseBoolBuildArg)</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)warnAsError $(ArcadeFalseBoolBuildArg)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(SourceBuildUseMonoRuntime)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)usemonoruntime</InnerBuildArgs>
+      <InnerBuildArgs Condition="'$(CrossBuild)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)cross</InnerBuildArgs>
       <!-- TODO: This parameter is only available on the Unix script. Intentional? -->
       <InnerBuildArgs Condition="'$(OS)' != 'Windows_NT'">$(InnerBuildArgs) --outputrid $(TargetRid)</InnerBuildArgs>
       <!-- PackageOS and ToolsOS control the rids of prebuilts consumed by the build.
-- 
2.44.0.windows.1

