From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Adeel Mujahid <3840695+am11@users.noreply.github.com>
Date: Wed, 26 Jun 2024 11:04:39 +0300
Subject: [PATCH] Also pass NETCoreSdkRuntimeIdentifier to runtime.proj

---
 eng/Subsets.props | 1 +
 1 file changed, 1 insertion(+)

diff --git a/eng/Subsets.props b/eng/Subsets.props
index 68569344a6e..532928f67da 100644
--- a/eng/Subsets.props
+++ b/eng/Subsets.props
@@ -292,6 +292,7 @@
                             HostOS=$(HostOS);
                             TargetOS=$(TargetOS);
                             RuntimeIdentifier=$(RuntimeIdentifier);
+                            NETCoreSdkRuntimeIdentifier=$(NETCoreSdkRuntimeIdentifier);
                             NETCoreSdkPortableRuntimeIdentifier=$(NETCoreSdkPortableRuntimeIdentifier);
                             PgoInstrument=false;
                             NoPgoOptimize=true;
