From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MichaelSimons <msimons@microsoft.com>
Date: Fri, 18 Feb 2022 15:11:39 +0000
Subject: [PATCH] Resolve NU1603 issue w/System.Reflection.Metadata during
 bootstrapping

---
 .../Microsoft.DotNet.PackageTesting.csproj                 | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/Microsoft.DotNet.PackageTesting/Microsoft.DotNet.PackageTesting.csproj b/src/Microsoft.DotNet.PackageTesting/Microsoft.DotNet.PackageTesting.csproj
index 824dc8b..07697f8 100644
--- a/src/Microsoft.DotNet.PackageTesting/Microsoft.DotNet.PackageTesting.csproj
+++ b/src/Microsoft.DotNet.PackageTesting/Microsoft.DotNet.PackageTesting.csproj
@@ -8,6 +8,13 @@
     <IsPackable>true</IsPackable>
   </PropertyGroup>
 
+  <!-- During bootstrapped source-builds, the following situation can occur:
+  Microsoft.CodeAnalysis.Common z.b.c depends on System.Reflection.Metadata (>= x.y.z) but System.Reflection.Metadata x.y.z was not found.
+  An approximate best match of System.Reflection.Metadata x.y.(z+1) was resolved. -->
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <NoWarn>$(NoWarn);NU1603</NoWarn>
+  </PropertyGroup>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildVersion)" Publish="false" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" Publish="false" />
