From 3c008b2c3f04f2d0bd04ae0c1f7d56a64141842c Mon Sep 17 00:00:00 2001
From: Michael Simons <msimons@microsoft.com>
Date: Mon, 16 Aug 2021 16:33:33 +0000
Subject: [PATCH] Add toolset zip as source build intermediate

---
 eng/SourceBuild.props | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index 8bc7785fb..539a1490a 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -11,4 +11,14 @@
     </PropertyGroup>
   </Target>
 
+  <Target Name="GetSdkCategorizedIntermediateNupkgContents"
+          BeforeTargets="GetCategorizedIntermediateNupkgContents">
+    <ItemGroup>
+      <!--
+        Add the internal toolset zip required by dotnet/installer.
+      -->
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)NonShipping\dotnet-toolset-internal-*.zip" />
+    </ItemGroup>
+  </Target>
+
 </Project>
-- 
2.29.2

