From 1cb584fab0d95a87418b884856c12ccc55ea0a95 Mon Sep 17 00:00:00 2001
From: dseefeld <dseefeld@microsoft.com>
Date: Fri, 10 Sep 2021 13:47:05 +0000
Subject: [PATCH] Exclude PdbTest project from source-build

---
 src/PdbTestResources/PdbTestResources.csproj | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/PdbTestResources/PdbTestResources.csproj b/src/PdbTestResources/PdbTestResources.csproj
index 1376ddf..21f45b3 100644
--- a/src/PdbTestResources/PdbTestResources.csproj
+++ b/src/PdbTestResources/PdbTestResources.csproj
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard1.3</TargetFramework>
     <IsShipping>false</IsShipping>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
   <ItemGroup>
     <Compile Remove="Resources\**\*.cs" />
@@ -18,4 +19,4 @@
       <LogicalName>SourceLink.pdb</LogicalName>
     </EmbeddedResource>
   </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
-- 
2.31.1

