From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MichaelSimons <msimons@microsoft.com>
Date: Mon, 5 Dec 2022 23:35:39 +0000
Subject: [PATCH] Disable package validation during source-build

---
 eng/SourceBuild.props | 1 +
 1 file changed, 1 insertion(+)

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index a034f88679..f3923138f8 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -55,6 +55,7 @@
 
     <PropertyGroup>
       <InnerBuildArgs>$(InnerBuildArgs) /p:SourceBuildRuntimeIdentifier=$(TargetRuntimeIdentifier)</InnerBuildArgs>
+      <InnerBuildArgs>$(InnerBuildArgs) /p:EnablePackageValidation=false</InnerBuildArgs>
     </PropertyGroup>
   </Target>
 
