From 4ca12d74308dceface792dad959bd4a7eda0b1b4 Mon Sep 17 00:00:00 2001
From: MichaelSimons <msimons@microsoft.com>
Date: Thu, 30 Sep 2021 22:37:24 +0000
Subject: [PATCH] Enable doc generation in source-build

---
 Build.proj | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Build.proj b/Build.proj
index bff94b97a7a..5f47e869fb6 100644
--- a/Build.proj
+++ b/Build.proj
@@ -10,7 +10,7 @@
   <Import Project="$(RepositoryEngineeringDir)SubsetValidation.targets" />
 
   <!-- Upfront restore hooks -->
-  <Import Project="$(RepositoryEngineeringDir)restore\docs.targets" Condition="'$(DotNetBuildFromSource)' != 'true'" />
+  <Import Project="$(RepositoryEngineeringDir)restore\docs.targets" />
   <Import Project="$(RepositoryEngineeringDir)restore\optimizationData.targets" Condition="'$(DotNetBuildFromSource)' != 'true'" />
 
   <Target Name="BuildLocalTasks"
-- 
2.29.2

