From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Rummel <crummel@microsoft.com>
Date: Wed, 21 Sep 2022 11:38:27 -0500
Subject: [PATCH] Patch out 'no entrypoint' warning as a workaround.

---
 src/Compiler/AbstractIL/ilwrite.fs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Compiler/AbstractIL/ilwrite.fs b/src/Compiler/AbstractIL/ilwrite.fs
index 983159604..adc338bc1 100644
--- a/src/Compiler/AbstractIL/ilwrite.fs
+++ b/src/Compiler/AbstractIL/ilwrite.fs
@@ -3143,7 +3143,6 @@ let generateIL (
             if isDll then 0x0
             else getUncodedToken (if epHere then TableNames.Method else TableNames.File) tok
         | None ->
-            if not isDll then dprintn "warning: no entrypoint specified in executable binary"
             0x0
 
     let pdbData =
