From e5fa6b10f5e4bcb14e3930e4eeae7181765529ee Mon Sep 17 00:00:00 2001
From: Chris Rummel <crummel@microsoft.com>
Date: Thu, 21 Oct 2021 16:29:32 -0500
Subject: [PATCH] Also build for netstandard2.0 to support NuGet.

---
 .../Microsoft.Web.XmlTransform.csproj                           | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Microsoft.Web.XmlTransform/Microsoft.Web.XmlTransform.csproj b/src/Microsoft.Web.XmlTransform/Microsoft.Web.XmlTransform.csproj
index 7cb81d1..f272d7f 100644
--- a/src/Microsoft.Web.XmlTransform/Microsoft.Web.XmlTransform.csproj
+++ b/src/Microsoft.Web.XmlTransform/Microsoft.Web.XmlTransform.csproj
@@ -3,7 +3,7 @@
   <Import Project="$(RepositoryEngineeringDir)\Package.props" />
   
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
     <PlatformTarget>AnyCPU</PlatformTarget>
     <IsPackable>true</IsPackable>
     <IsShipping>false</IsShipping>
-- 
2.31.1

